telegram/tests/test_telegram.py

```diff
@@ -0,0 +1,344 @@
+            'chat': default_channel,
+            'text': 'Hi from Telegram channel!',
+        }
+
+        message = self.transport.translate_inbound_message(inbound_msg)
+        self.assertEqual(inbound_msg['text'], message['content'])
+        self.assertEqual(self.bot_username, message['to_addr'])
+        self.assertEqual(default_channel['id'], message['from_addr'])
+
+    def test_translate_inbound_message_from_user(self):
+        inbound_msg = {
+            'message_id': 'Message from Telegram user',
+            'chat': 'Random chat',
+            'text': 'Hi from Telegram user!',
+            'from': self.default_user,
+        }
+
+        message = self.transport.translate_inbound_message(inbound_msg)
+        self.assertEqual(inbound_msg['text'], message['content'])
+        self.assertEqual(self.bot_username, message['to_addr'])
+        self.assertEqual(self.default_user['id'], message['from_addr'])
+
+    @inlineCallbacks
+    def test_inbound_update(self):
+        update = {
+            'update_id': 'update_id',
+            'message': {
+                'message_id': 'msg_id',
+                'from': self.default_user,
+                'chat': {
+                    'id': 'chat_id',
+                    'type': self.PRIVATE
+                },
+                'date': 1234,
+                'text': 'Incoming message from Telegram!',
+            }
+        }
+
+        with LogCatcher(message='TelegramTransport') as lc:
+            res = yield self.helper.mk_request(_method='POST',
+                                               _data=json.dumps(update))
+            self.assertEqual(res.code, http.OK)
+            [log] = lc.messages()
+            self.assertEqual(
+                log,
+                'TelegramTransport receiving inbound message from %s to %s' %
+                (self.default_user['id'], self.bot_username)
+            )
+
+        [msg] = yield self.helper.wait_for_dispatched_inbound(1)
+        self.assertEqual(msg['to_addr'], self.bot_username)
+        self.assertEqual(msg['from_addr'], self.default_user['id'])
+        self.assertEqual(msg['content'], update['message']['text'])
+        self.assertEqual(msg['transport_type'],
+                         self.transport.transport_type)
+        self.assertEqual(msg['transport_name'],
+                         self.transport.transport_name)
+
+    @inlineCallbacks
+    def test_inbound_non_message_update(self):
+        update = json.dumps({
+            'update_id': 'update_id',
+            'object': 'This is not a message...',
+        })
+        d = self.helper.mk_request(_method='POST', _data=update)
+
+        with LogCatcher(message='message') as lc:
+            res = yield d
+            [log] = lc.messages()
+            self.assertEqual(log, 'Inbound update does not contain a message')
+        self.assertEqual(res.code, http.OK)
+
+    @inlineCallbacks
+    def test_inbound_non_text_message(self):
```

Comment:
Following on from my previous comment, this test's docstring could be "If we receive a non-text inbound message, we log what happened and ignore the message."